lib/functions/aemcompleteupload.js

```diff
@@ -61,7 +62,11 @@ class AEMCompleteUpload extends AsyncGeneratorFunction {
                 const form = new URLSearchParams();
                 form.append("fileName", transferAsset.target.filename);
                 form.append("fileSize", transferAsset.metadata.contentLength);
-                form.append("mimeType", transferAsset.metadata.contentType);
+                if (transferAsset.metadata.contentType) {
+                    form.append("mimeType", transferAsset.metadata.contentType);
+                } else {
+                    form.append("mimeType", MIMETYPE.APPLICATION_OCTET_STREAM);
+                }
```

Comment:
It is unlikely in normal operation for `transferAsset.metadata.contentType` to be null or undefined, since you have to call `AEMInitiateUpload` first which is guaranteed to set it. However, just in case no `contentType` is provided we pass in the default. Passing in null or undefined here results in a string `null` or `undefined`.

Reply:
Would there be the need to do some more verifications on the content type? For instance, that it's not an empty string (n spaces) or a blank character (like `\n` or `\t` for instance)? I wouldn't do a full-blown validation since I'd expect custom mimetype here...

Reply:
It's supposed to already be validated, we have checks in the `AssetMetadata` constructor: https://github.com/adobe/node-httptransfer/blob/aemupload_mimetype_undefined/lib/asset/assetmetadata.js#L34-L36

Maybe they can be expanded, but I'm not sure if that's a good idea since we'll run the risk to reject something a customer needs but is technically invalid.